bolt-sidecar/bin/sidecar.rs

```diff
@@ -138,3 +142,49 @@ async fn main() -> eyre::Result<()> {
 
     Ok(())
 }
+
+/// Filters the proposer duties and returns the validator index for a given slot
+/// if it doesn't exists then returns 0 by default.
+pub fn find_validator_index_for_slot(
+    validator_indexes: &[u64],
+    proposer_duties: &[ProposerDuty],
+    slot: u64,
+) -> u64 {
+    for duty in proposer_duties {
+        if duty.slot == slot && validator_indexes.contains(&(duty.validator_index as u64)) {
+            return duty.validator_index as u64;
+        }
+    }
+    0
+}
```

Comment:
A more idiomatic approach to do this (imo) is the following:
```rs
proposer_duties.iter().find(|duty| duty.slot == slot && ... )
```
There returns an `Option<ProposerDuty>` which is more flexible compared to returning `0` by default (maybe who uses this would like a different value). If the result is not `None` then we the caller can easily extract the validator index.
Wydt?

Reply:
yes I agree, I have updated the fn now in a similar way